src/ball.rs

```diff
@@ -40,17 +71,33 @@ impl Ball {
     /// translated to (0.0, 0.0).
     ///
     pub fn update(&mut self, input_axes: Vec2, delta_time: f32) {
-        // Update the ball's position
+        let mut rng = rand::thread_rng();
+
+        if self.time_running >= self.velocity_change_time_at {
+            let radians: f32 = rng.gen_range(0.0..2.0 * f32::consts::PI);
+            let (new_x, new_y) = (radians.cos(), radians.sin());
+            let hyp = f32::sqrt(self.velocity.x.powi(2) + self.velocity.y.powi(2));
```

Comment:
do they have a length function that does this?

Reply:
https://docs.rs/egui/latest/egui/struct.Vec2.html#method.length

Reply:
Just integrated this solution, please check if this satisfy your requirement.

Reply:
very nice!